src/platform_impl/linux/x11/mod.rs

```diff
@@ -269,7 +269,8 @@ impl<T: 'static> EventLoop<T> {
 
         // Create the X11 event dispatcher.
         let source = X11Source::new(
-            xconn.xcb_connection().as_raw_fd(),
+            // SAFETY: xcb owns the FD and outlives the source.
```

Comment:
r @notgull . You could probably do something smarter than that, but I just did a conservative change.

Reply:
I guess `XKBConnection` doesn't implement `AsFd`? It probably should.

Then it could be contained in the `Generic` directly. After wrapping with ref-counting if another reference is needed outside the source.

Reply:
well, upstream only given `c_int` as a fd iirc.

Reply:
https://github.com/psychon/x11rb/pull/889 adds an `AsFd` implementation here.

Reply:
This works for now, until the PR above is published.